File: C++/Cristian Maldonado/Minimum_Absolute_Sum_Pair.cpp
#include<iostream>
using namespace std;

void minAbsSumPair(int arr[], int n){
	int l, r, min_sum,sum =0, min_l, min_r;
	if(n < 2){
		cout << "Array no tiene la cantidad de valores necesarios"<<endl; 
		return;
	}
	min_l = 0;
	min_r = 1;
	min_sum = arr[0] + arr[1];
	for(l = 0; l < n; l++){
		for(r = l+1; r <n;r++){
			sum = arr[l]+arr[r];
			if(abs(min_sum) > abs(sum)){
				min_sum = sum;
				min_l = l;
				min_r = r;
			}
		}
	}
	cout << "Los dos elementos con la suma minima son: "<<arr[min_l]<<" y "<<arr[min_r]<<endl; 
}

int main (int argc, char *argv[]) {
	int arr [] = {1,60,-10,70,-80,85};
	minAbsSumPair(arr,sizeof(arr)/sizeof(0));
	return 0;
}

File: PY/Cristian Maldonado/Minimum_Absolute_Sum_Pair.py
def minAbsSumPair(arr, n):
    if (n < 2):
        print("Array no tiene la cantidad de valores necesarios")
        return
    sum = 0
    min_x = 0
    min_r = 1
    min_sum = arr[0] + arr[1]
    for x in range(0, n):
        for r in range(x + 1, n):
            sum = arr[x] + arr[r]
            if (abs(min_sum) > abs(sum)):
                min_sum = sum
                min_x = x
                min_r = r
    print("Los dos elementos con la suma minima son: (" + str(arr[min_x]) + ", " + str(arr[min_r]) + ")")


arr = [1, 60, -10, 70, -80, 85]
minAbsSumPair(arr, len(arr))
